### PR TITLE
Docs fix

### PR DESCRIPTION
**Pull request by zainsohail04**
With regards to #107, I have fixed the issue occuring with readthedocs. It was related to the requirements.txt in docs being out of date. I have now set up a action that will autogenerate a requirements.txt from the poetry.lock at each push to [main].

To review, you can check this https://sed.readthedocs.io/en/docs-fix/ (the build is hidden in the menu as it's just for test purposes). 

Second thing I believe would be useful is to have a seperate sed email address that can be used to access readthedocs etc.

### CODE REVIEW

**Comment by zainsohail04** [comment]
Not sure where this error popped up from:
`tests/calibrator/test_momentum.py:242: KeyError`

**Comment by rettigl** [comment]
> Not sure where this error popped up from: `tests/calibrator/test_momentum.py:242: KeyError`

This is a random test failure that I fixed in PR #143 . Just re-run the tests and it should disappear.

File: .github/workflows/generate-requirements.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    branches: [main]
+
+jobs:
+  generate-requirements:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Poetry
+        run: sudo apt install poetry
+      - name: Export requirements.txt
+        run: poetry export --without-hashes --format=requirements.txt -o docs/requirements.txt -E docs -E notebook
```
Comment (rettigl) [inline .github/workflows/generate-requirements.yml:14]: Did you test this somehow? Will this work on pull-request merge?

**Review by rettigl — APPROVED** [review]
LGTM, Not 100% sure the workflow does fire on PR merges, but let's give it a try.

File: .github/workflows/generate-requirements.yml (hunk 1 above)
Comment (zainsohail04) [reply .github/workflows/generate-requirements.yml:14]: I can't really test it locally but on PR merge, the on push should work. Let's try.

**Pull request merged**